### PR TITLE
Check cops online for robbing AND surrender action

### PR DESCRIPTION
**Pull request by clackd1amond**

### CODE REVIEW

**Comment by Heavybob** [comment]
Thanks @clackd1amond for merging surrender action and doing this bug fix.

It is still possible to bolt cut the research box which contains the fed box.

[16015.5,16958,0.00978851]
5e392400# 499787: research_house_v1_f.p3d

**Comment by Heavybob** [comment]
Approve commit :)

File: Altis_Life.Altis/core/functions/fn_keyHandler.sqf [hunk 1]
```diff
@@ -109,6 +109,17 @@ switch (_code) do {
 		};
 	};
 
+	//Surrender... Tab Key
+	case 15: {
+		if(!_alt && !_ctrlKey && !_shift && !dialog && !(player GVAR ["restrained",false]) && (animationState player) != "Incapacitated" && !life_istazed) then {
+			if (player GVAR ["surrender", false]) then {
```
Comment (sargken) [inline Altis_Life.Altis/core/functions/fn_keyHandler.sqf]: If someone can fill me in more, what is the ["surrender", false], false stand for. Is that the broadcast setting, or is it checking for it?

**Comment by tkcjesse** [comment]
@sargken It's used to change the animation states. It's just setting the variable then the action script for surrender gets that variable to decide what animation state to do when you hit the key for a 2nd/3rd/4th/etc time.

**Comment by sargken** [comment]
@tkcjesse I know that, what does the false specifically do in the SVAR statement. Is the the Boolean for broadcast, or is that what its looking for.

**Comment by Jawshy** [comment]
@clackd1amond, you should also specify the surrender key under "controls" in the briefing.sqf.

**Comment by Heavybob** [comment]
I'll add a check to see if player in veh. It gets a little silly.

**Comment by Jawshy** [comment]
I would like to suggest that in **fn_setupActions.sqf** line 14 be changed from

>		!isNull cursorTarget && player distance cursorTarget < 3.5 && isPlayer cursorTarget && animationState cursorTarget == "Incapacitated" && !(cursorTarget getVariable["robbed",FALSE]) ']];

to

>		!isNull cursorTarget && player distance cursorTarget < 3.5 && isPlayer cursorTarget && animationState cursorTarget == "Incapacitated" OR animationState cursorTarget == "AmovPercMstpSnonWnonDnon_AmovPercMstpSsurWnonDnon" && !(cursorTarget getVariable["robbed",FALSE]) ']];

Difference
>		OR animationState cursorTarget == "AmovPercMstpSnonWnonDnon_AmovPercMstpSsurWnonDnon" 

This means that players that have surrendered can also be robbed, but note that there is no check to see if the robber has a weapon. I.e. A surrendered player can be robbed by a weapon-less player.

**Comment by danielstuart14** [comment]
Going to test and commit on thursday. I promise guys.

**Comment by Jawshy** [comment]
Bump: The "Check cops online" part is really good.

**Comment by danielstuart14** [comment]
https://github.com/ArmaLife/Altis/commit/64d87e3173ab7d9985a1ebb23d93bac757577e31

**Pull request closed without merging**